### PR TITLE
client: Add CreateInOrder method to client.KeysAPI

### PR DESCRIPTION
**Pull request by ecnahc515**
Allows creating nodes within a given directory with atomically increasing
keys

This Succeeds #2410

### CODE REVIEW

File: client/keys.go [hunk 1]
```diff
@@ -133,6 +136,14 @@ type WatcherOptions struct {
 	Recursive bool
 }
 
+type CreateInOrderOptions struct {
```
Comment (xiang90) [inline client/keys.go]: this options seem to be unnecessary. Why not just put TTL in the arg list?

Comment (ecnahc515) [inline client/keys.go]: This is more future proof, we can add more options into this later if necessary

File: client/keys.go [hunk 2]
```diff
@@ -294,6 +305,25 @@ func (k *httpKeysAPI) Create(ctx context.Context, key, val string) (*Response, e
 	return k.Set(ctx, key, val, &SetOptions{PrevExist: PrevNoExist})
 }
 
+func (k *httpKeysAPI) CreateInOrder(ctx context.Context, dir, val string, opts *CreateInOrderOptions) (*Response, error) {
+	act := &createInOrderAction{
+		Prefix: k.prefix,
+		Dir:    dir,
+		Value:  val,
+	}
```
Comment (bakins) [inline client/keys.go]: Why pass in `dir` and `val` and have them in struct? Perhaps confusing to users?

File: client/keys.go (hunk 1 above)
Comment (xiang90) [inline client/keys.go]: ok.

**Comment by xiang90** [comment]
LGTM. Please address @bakins's concern though.

**Comment by bakins** [comment]
LGTM

File: client/keys_test.go [hunk 3]
```diff
@@ -822,7 +923,7 @@ func TestHTTPKeysAPIWatcherAction(t *testing.T) {
 	}
 }
 
-func TestHTTPKeysAPISetAction(t *testing.T) {
```
Comment (bcwaldon) [inline client/keys_test.go]: It appears that this change should not have been made.

**Pull request merged**